File: problems/2869-minimum-operations-to-collect-elements/solution.java
import java.util.List;
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int minOperations(List<Integer> nums, int k) {
        Set<Integer> collected = new HashSet<>();
        int operations = 0;
        for (int i = nums.size() - 1; i >= 0; i--) {
            operations++;
            int num = nums.get(i);
            if (num >= 1 && num <= k) {
                collected.add(num);
            }
            if (collected.size() == k) {
                return operations;
            }
        }
        return operations;
    }
}

File: problems/2869-minimum-operations-to-collect-elements/solution.c
int minOperations(int* nums, int numsSize, int k) {
    int count = 0;
    int operations = 0;
    int collected[51] = {0};
    for (int i = numsSize - 1; i >= 0; i--) {
        operations++;
        if (nums[i] >= 1 && nums[i] <= k && collected[nums[i]] == 0) {
            collected[nums[i]] = 1;
            count++;
        }
        if (count == k) {
            return operations;
        }
    }
    return operations;
}